### ISSUE TITLE
Settings: ARIA labeling & Localization problem in the new settings editor

### ISSUE THREAD

**Issue by qcz** [Microsoft/vscode#59147]
There is a new piece of translatable text (settingsContextMenuAriaShortcut) for AIRA in `settingsEditor2.ts`:
https://github.com/Microsoft/vscode/blob/ab13118d54555dd5462ed0ed28b20a19b3a15f2c/src/vs/workbench/parts/preferences/browser/settingsEditor2.ts#L277-L278
However it is using string concatenation to append parts of the text which makes translation hard as the word order may differ in different languages. For example in Hungarian some parts of the translated text goes after the shortcut which is impossible to do currently. Furthermore if somebody translates without checking the source code of Code, they may misunderstand the meaning of the 'Press' at the end.

Placeholders should be used to insert the shortcut into the text, like here:
https://github.com/Microsoft/vscode/blob/ab13118d54555dd5462ed0ed28b20a19b3a15f2c/src/vs/workbench/parts/preferences/browser/settingsEditor2.ts#L1166

FYI @cleidigh (author of 26b228d3dca0438549917f5123cb34121d0f186d), @roblourens

**Comment by roblourens** [comment]
Good catch, @cleidigh can you fix it?

**Comment by cleidigh** [comment]
@qcz 
@roblourens 
This creates a bit of a Catch-22, we have some dynamic parts and user tips I'm really not sure what's the best answer.  Do we have any other close examples?

**Comment by cleidigh** [comment]
I think I'm being dense
@qcz 
If we use   
localize("shortcut prefix {0}", shortcutLabel)
Can translations create translation text both before and after the parameter?

**Comment by qcz** [comment]
@cleidigh: Exactly, you create these like format strings in programming languages, and translators (who are used to them) put placeholders where needed.

Prefix and postfix strings (which you proposed in your second comment) may work if there is only one parameter (not if there are more, because they may be swapped), but they are actively discouraged when doing internationalization. Furthermore Transifex, the platform we use for localization of VS Code is not very good at showing contextual information for translators (they are visible, but not prominently), so an empty string most likely would confuse translators.

**Comment by cleidigh** [comment]
should be good now...

**Comment by qcz** [comment]
@cleidigh: It looks great! Thanks!

**Comment by cleidigh** [comment]
@roblourens 
Related item: I'm going to do a cleanup pass on the ARIA labeling code now that things have settled a bit, remove duplicate code/optimize et cetera.

**Comment by cleidigh** [comment]
@roblourens 
I did not realize it was code freeze day.  I better push these into October to be safe.

**Comment by cleidigh** [comment]
Absolutely, I'm almost done with a new interface where you can pass options, details, labeling, extras like defaults as well as Booleans indicating markdown content.  This also reduces the calls required further improving performance.  

I don't think I can put these in for September because it's closed , correct?

Any comments on below:

```ts

export interface ISelectBoxOptionEntry {
	optionEntry: string;
	optionEntryIsMarkdown?: boolean;
	optionEntrySecondary?: string;
	optionEntrySecondaryIsMarkdown?: boolean;
	optionEntryDetails?: string;
	optionEntryDetailsIsMarkdown?: boolean;
	isSelected?: boolean;
	isDefault?: boolean;
	isDisabled?: boolean;
	optionEntryAriaLabel?: string;
	optionEntryAriaLabeledby?: string;

}
```

**Comment by roblourens** [comment]
I'll respond on #58236
